server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -233,6 +236,7 @@ public DruidCoordinator(
     this.coordLeaderSelector = coordLeaderSelector;
 
     this.segmentCompactor = new DruidCoordinatorSegmentCompactor(indexingServiceClient);
+    this.dataSourcesSnapshot = metadataSegmentManager.getDataSourcesSnapshot();
```

Comment:
This shouldn't be in the constructor, the snapshot should be fetched at the start of every ooordinator loop iteration

Reply:
thanks, updated